canvas_sdk/tests/handlers/simple_api/test_tools.py

```diff
@@ -0,0 +1,80 @@
+from collections.abc import Callable
+from typing import TypeVar
+
+import pytest
+
+from canvas_sdk.handlers.simple_api.tools import CaseInsensitiveMultiDict, MultiDict
+
+T = TypeVar("T")
+
+
+@pytest.mark.parametrize(
+    argnames="func,expected_value",
+    argvalues=[
+        (lambda m: m["b"], 2),
+        (lambda m: m["a"], 1),
+        (lambda m: len(m), 2),
+        (lambda m: next(iter(m.items())), ("a", 1)),
+        (lambda m: "a" in m, True),
+        (lambda m: "d" in m, False),
+        (lambda m: m.get("a"), 1),
+        (lambda m: m.get("d", 4), 4),
+        (lambda m: m.get("d"), None),
+        (lambda m: m.get_list("a"), [1, 3]),
+        (
+            lambda m: [(k, v) for k, v in m.items()],
+            [("a", 1), ("b", 2)],
+        ),
+        (
+            lambda m: [(k, v) for k, v in m.multi_items()],
+            [("a", 1), ("b", 2), ("a", 3)],
+        ),
+        (lambda m: list(m.keys()), ["a", "b"]),
+        (lambda m: list(reversed(m)), ["b", "a"]),
+        (lambda m: list(m.values()), [1, 2]),
+        (lambda m: m == MultiDict((("a", 1), ("b", 2), ("a", 3))), True),
+        (lambda m: m != MultiDict((("a", 1), ("b", 2))), True),
+    ],
+    ids=[
```

Comment:
easier to keep the ids in sync if you use `params_from_dict` like this: https://github.com/canvas-medical/canvas-plugins/blob/43335cdda293442084d0036d55d557fbbf3bc0ac/plugin_runner/tests/test_sandbox.py#L625-L670